Source/control.cpp

```diff
@@ -563,6 +561,67 @@ void RemoveGold(int pnum, int goldIndex)
 	dropGoldValue = 0;
 }
 
+bool GetSpellListSelection(spell_id &pSpell, spell_type &pSplType)

```

Comment:
Does this whole thing replace some other part of the code, or can it? It seams a lot of code for what used to just be `pSpell == SPL_INVALID` :)

Nice that it gets rid of some globals though :+1:

Reply:
It also gets you the valid spells. 😉 

One possibility could be to make a new functions that returns a list of all spells and there rendering positions.
This could be used in both rendering and selection logic.

Reply:
i tried this approach. Take a look at the new commit. 🙂